src/core/version_neg.c

```diff
@@ -171,217 +171,94 @@ QuicVersionNegotiationExtGenerateCompatibleVersionsList(
-    if (BufferLength < sizeof(uint32_t)) {
-        QuicTraceLogConnError(
-            ServerVersionNegotiationInfoDecodeFailed1,
-            Connection,
-            "Server version negotiation info too short to contain Negotiated Version (%hu bytes)",
-            BufferLength);
-        return QUIC_STATUS_INVALID_PARAMETER;
-    }
-    CxPlatCopyMemory(&ServerVNI->NegotiatedVersion, Buffer, sizeof(ServerVNI->NegotiatedVersion));
-    Offset += sizeof(uint32_t);
-
-    if (!QuicVarIntDecode(BufferLength, Buffer, &Offset, &ServerVNI->SupportedVersionCount)) {
-        QuicTraceLogConnError(
-            ServerVersionNegotiationInfoDecodeFailed2,
-            Connection,
-            "Server version negotiation info too short to contain Supported Version count (%hu bytes)",
-            BufferLength);
-        return QUIC_STATUS_INVALID_PARAMETER;
-    }
-
-    if (ServerVNI->SupportedVersionCount * sizeof(uint32_t) > (uint64_t)BufferLength - Offset) {
-        QuicTraceLogConnError(
-            ServerVersionNegotiationInfoDecodeFailed3,
-            Connection,
-            "Server version negotiation info too short to contain Supported Versions list (%hu bytes vs. %llu bytes)",
-            BufferLength,
-            ServerVNI->SupportedVersionCount * sizeof(uint32_t));
-        return QUIC_STATUS_INVALID_PARAMETER;
-    }
-
-    if (ServerVNI->SupportedVersionCount > 0) {
-        ServerVNI->SupportedVersions = (uint32_t*)(Buffer + Offset);
-        Offset += (uint16_t)(ServerVNI->SupportedVersionCount * sizeof(uint32_t));
-    } else {
-        ServerVNI->SupportedVersions = NULL;
-        QuicTraceLogConnError(
-            ServerVersionNegotiationInfoDecodeFailed4,
-            Connection,
-            "Server version negotiation info has empty Supported Versions list");
-        return QUIC_STATUS_INVALID_PARAMETER;
-    }
-    if (Offset != BufferLength) {
-        QuicTraceLogConnError(
-            ServerVersionNegotiationInfoDecodeFailed5,
-            Connection,
-            "Server version negotiation info parsed less than full buffer (%hu bytes vs. %hu bytes",
-            Offset,
-            BufferLength);
-        return QUIC_STATUS_INVALID_PARAMETER;
-    }
-
-    QuicTraceLogConnInfo(
-        ServerVersionNegotiationInfoDecoded,
-        Connection,
-        "Server VNI Decoded: Negotiated Ver:%x Supported Ver Count:%llu",
-        ServerVNI->NegotiatedVersion,
-        ServerVNI->SupportedVersionCount);
-
-    QuicTraceEvent(
-        ConnServerSupportedVersionList,
-        "[conn][%p] Server VNI Supported Version List: %!VNL!",
-        Connection,
-        CASTED_CLOG_BYTEARRAY(ServerVNI->SupportedVersionCount * sizeof(uint32_t), ServerVNI->SupportedVersions));
+        Source,
+        CASTED_CLOG_BYTEARRAY(VersionInfo->OtherVersionsCount * sizeof(uint32_t), VersionInfo->OtherVersions));
 
     return QUIC_STATUS_SUCCESS;
 }
 
 _IRQL_requires_max_(DISPATCH_LEVEL)
 _Success_(return != NULL)
 const uint8_t*
```

Comment:
We should annotate that the return is of length `VNInfoLength`.

Reply:
Is there an example of that in our code?